chiselFrontend/src/main/scala/chisel3/core/UserModule.scala

```diff
@@ -38,17 +38,22 @@ abstract class UserModule(implicit moduleCompileOptions: CompileOptions)
 
   val compileOptions = moduleCompileOptions
 
+  private[chisel3] def namePorts(names: HashMap[HasId, String]): Unit = {
```

Comment:
Does this detect naming conflicts? Previously we didn't need to worry about this because (presumably) you can't get a conflict with member accessors, but given arbitrary strings, I think this is where you want to error out where you have two wires fighting for the same name

Reply:
Currently you can get conflicts using suggestName:
```scala
class TestModule extends Module {
  val io = IO(new Bundle { })
  val w = WireInit(10.U).suggestName("foo")
  val w2 = WireInit(10.U).suggestName("foo")
}
```
gives
```
circuit TestModule :
  module TestModule :
    input clock : Clock
    input reset : UInt<1>
    output io : {}

    wire foo : UInt
    foo <= UInt<4>("h0a")
    wire foo_1 : UInt
    foo_1 <= UInt<4>("h0a")
```
This PR doesn't change anything about that.

That being said, there definitely should be some tests illustrating how conflicts *are* resolved. TLDR, ports always win over things like wires, but _# is introduced when ports conflict as well.

Reply:
I think it makes sense to check for conflicts in namePorts and error out if one is detected? I don't know if it makes sense to refactor namespace to do that checking, but that's definitely out of scope for this,

Reply:
I think a stricter `.Name` API could error. I don't think we should change `.suggestName` to error since it's name suggests it won't, but maybe I can make it error just for ports? I'll look into how hard that is to do

Reply:
Yeah, I think it should error for ports. We don't have to worry about wires until suggestName is replaced, but I think that a particularly nasty case could be a conflict between a suggested name and a val name, and one of them (probably based on call order) will unexpectedly turn into the wrong thing in the generated verilog.

Reply:
It now errors for ports getting suggested (or otherwise named) to the same thing